pkg/server/server.go

```diff
@@ -465,14 +468,17 @@ func (s *ProxyServer) serveRecvBackend(stream agent.AgentService_ConnectServer,
 			} else {
 				err := client.send(pkt)
 				s.PendingDial.Remove(resp.Random)
-				if err != nil {
+				if resp.Error != "" {
+					klog.Warningf("<<< DIAL_RSP received error: %v", resp.Error)
+					break
```

Comment:
We shouldn't break here. One dial failed, but there are other streams on this backend, we should keep processing these other streams.

Reply:
This breaks out of the case statement, not the for loop (Only intended to skip L478-481). It's a bit ambiguous which is why I considered changing this to use if statements instead of case. Let me know your thoughts.

Reply:
Got it. I think it's ok leaving it as "switch/case".